### ISSUE TITLE
corrupted mean img

### ISSUE THREAD

**Issue by or-naim** [MouseLand/suite2p#290]
Hello,
First of all thank you so much for your hard work! 

I have a TIFF file of 1P data with 220 frames that I created in MATLAB from a 3d matrix MATfile, that looks like this:
![image](https://user-images.githubusercontent.com/59688604/75440238-114bbf80-5964-11ea-8de7-db45c35c4be1.png)

After I run suite2p Python version on this data, the mean img that I get looks very unclear and fuzzy:
![image](https://user-images.githubusercontent.com/59688604/75440679-e2821900-5964-11ea-8f3d-95d1296774d8.png)

I have tried to change the parameters to suit the 1P data but I do not think the problem is here (and also it didn't change the results). 

On top of this, even though I am getting the above corrupted mean img, the program does recognize some cells (14), but weirdly enough shows that the activity of all of them is off after frame 120 (although the 120-220 th frames are completely fine and seem to be quite similar to the first ones). 
![image](https://user-images.githubusercontent.com/59688604/75441739-deef9180-5966-11ea-8b81-9d069930f0c1.png)

Any suggestions?

**Comment by carsen-stringer** [comment]
when you make a tiff make sure you make an "int16" tiff so that the dynamic range is larger, e.g. something like this would work where X is your data matrix
~~~
tif = (X - np.min(X)) / (np.max(X) - np.min(X)) # to 0-1 range
tif *= 15000
tif = np.int16(tif)
~~~

**Comment by or-naim** [comment]
Thanks!
So the first step (0-1 range) was already made on the MATLAB data above. 
I tried to multiply the matrix by 15000 and save it as uint16, but the problem remains.

maybe the problem is with the MATLAB function that I use? (imwrite)
those are the code lines from MATLAB ('tr4' is the 3d data matrix, all the numbers are already above zero so I only normalized the shifted data to 1) :

`
% the first tiff object
s = tr4(:, :, 1);
s = s/max(s(:));
s = s * 15000;
imwrite(uint16(s), 'frame4.Tiff');

% the rest tiff's objects appended to the first file
for i = 2:220
    s = tr4(:, :, i);
    s = s/max(s(:));
    s = s * 15000;
    imwrite(uint16(s), 'frame4.Tiff','WriteMode','append');
end`

**Comment by carsen-stringer** [comment]
Is tr4 a float or an int? when you imagesc(s) in matlab it looks fine after the processing you do?

**Comment by or-naim** [comment]
tr4 is a float at the beginning, but the imwrite(uint16..) command automatically turns it to int. 
Yes, with imagesc(s) it looks perfect. see the attached picture below. It also looks the same even before the processing.

figure;imagesc(s);colormap(gray);
=
figure;imagesc(tr4(:, :, 1));colormap(gray); 

same result:

![imagesc](https://user-images.githubusercontent.com/59688604/79008074-43586000-7b65-11ea-8954-f4c093d08192.png)
 
I am also attaching the parameters. maybe there is a problem there:

![parameters](https://user-images.githubusercontent.com/59688604/79008335-d5606880-7b65-11ea-806e-23f67b1099b1.png)

Two more things - maybe the number of frames (220) may cause the problem? or maybe the total activity of the cells that oddly turns off after frame 120 can give us some hint?

![Inkedscreen suite2p_LI](https://user-images.githubusercontent.com/59688604/79008927-11e09400-7b67-11ea-881c-320861aae11e.jpg)

Many thanks!
-Or

**Comment by carsen-stringer** [comment]
hmm really no idea what's going on, can you send me a link to this file or email it to me?

**Comment by carsen-stringer** [comment]
I don't think I got an email about this, I'm going to close this issue for now

**Comment by or-naim** [comment]
Hey Carsen, 
It's been a while, but I think I figured it out!

First of all, for an unknown reason the files was compressed, so I had to cancel it by writing the 'compression' scheme into the imwrite command, writing explicitly 'none' afterwards to cancel the compression (generally it suppose to be the default).

`imwrite(matrix, 'name_of_file.tiff','Compression','none')
`
The second problem was trying to convert it to int16. I don't know why but for some reason it worked only when I tried to use the raw data. 

Below are the matlab commands. Hopefully it will help others.

```
mm = raw_matrix(:,:,1)
I = mat2gray(mm,[0.8922 1.0946])
imwrite(I, 'nocompress.tiff','Compression','none')

for i = 2:220
   mm = raw_matrix(:,:,i)
   I = mat2gray(mm,[0.8922 1.0946])
   imwrite(I, 'nocompress.tiff','Compression','none','WriteMode','append') 
end
```

Thank you so much for your help!
